src/dispatch/plugins/dispatch_slack/incident/interactive.py

```diff
@@ -0,0 +1,1842 @@
+        assigner_email=user.email,
+        assignee_email=assignee_email,
+        assignee_role=assignee_role,
+        db_session=db_session,
+    )
+
+    if (
+        assignee_role == ParticipantRoleType.reporter
+        or assignee_role == ParticipantRoleType.incident_commander
+    ):
+        # we update the external ticket
+        incident_flows.update_external_incident_ticket(
+            incident_id=context["subject"].id, db_session=context["subject"].organization_slug
+        )
+
+    modal = Modal(
+        title="Assign Role", blocks=[Section(text="Assigning role... Success!")], close="Close"
+    ).build()
+    await client.views_update(view_id=body["view"]["id"], view=modal)
+
+
+app.view(
+    AssignRoleActions.submit,
+    middleware=[action_context_middleware, db_middleware, user_middleware, modal_submit_middleware],
+)(ack=ack_assign_role_submission_event, lazy=[handle_assign_role_submission_event])
+
+
+async def handle_engage_oncall_command(ack, respond, context, body, client, db_session):
+    """Handles the engage oncall command."""
+    await ack()
+    # TODO handle cases
+    if context["subject"].type == "case":
+        await respond(
+            text="Command is not currently available for cases.", response_type="ephemeral"
+        )
+        return
+
+    incident = incident_service.get(db_session=db_session, incident_id=context["subject"].id)
+
+    oncall_services = service_service.get_all_by_project_id_and_status(
+        db_session=db_session, project_id=incident.project.id, is_active=True
+    )
+
+    if not oncall_services.count():
+        await respond(
+            blocks=Message(
+                blocks=[
+                    Section(
+                        text="No oncall services have been defined. You can define them in the Dispatch UI at /services."
+                    )
+                ]
+            ).build()["blocks"],
+            response_type="ephemeral",
+        )
+        return
+
+    services = [{"text": s.name, "value": s.external_id} for s in oncall_services]
+
+    blocks = [
+        static_select_block(
+            label="Service",
+            action_id=EngageOncallActionIds.service,
+            block_id=EngageOncallBlockIds.service,
+            placeholder="Select Service",
+            options=services,
+        ),
+        Input(
+            block_id=EngageOncallBlockIds.page,
+            label="Page",
+            element=Checkboxes(
+                options=[PlainOption(text="Page", value="Yes")],
+                action_id=EngageOncallActionIds.page,
+            ),
+            optional=True,
+        ),
+    ]
+
+    modal = Modal(
+        title="Engage Oncall",
+        blocks=blocks,
+        submit="Engage",
+        close="Close",
+        callback_id=EngageOncallActions.submit,
+        private_metadata=context["subject"].json(),
+    ).build()
+
+    await client.views_open(trigger_id=body["trigger_id"], view=modal)
+
+
+@app.view(
```

Comment:
Why is `app.view` a decorator in this command function vs a function call in other commands?

Reply:
The function call version is using the "Lazy listeners (FaaS)" feature which @kevgliss introduced in this PR (https://github.com/Netflix/dispatch/pull/2767/commits/7471853389a763fbf516e113a90fb835ae934ef2) - I don't have the full context. But, I suspect it was just missed in the change. I've converted it to use the lazy version to stay consistent. Looks like it made it easier to send the `success` model.